python/src/lib.rs

```diff
@@ -15,18 +15,59 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use ballista::prelude::*;
+use datafusion::prelude::*;
+use datafusion_python::context::PySessionContext as DataFusionPythonSessionContext;
+use datafusion_python::utils::wait_for_future;
+use std::collections::HashMap;
+
 use pyo3::prelude::*;
-pub mod context;
 mod utils;
 
-pub use crate::context::PySessionContext;
-
 #[pymodule]
-fn pyballista_internal(_py: Python, m: Bound<'_, PyModule>) -> PyResult<()> {
+fn ballista_internal(_py: Python, m: Bound<'_, PyModule>) -> PyResult<()> {
     pyo3_log::init();
     // Ballista structs
-    m.add_class::<PySessionContext>()?;
+    m.add_class::<PyBallista>()?;
     // DataFusion structs
     m.add_class::<datafusion_python::dataframe::PyDataFrame>()?;
     Ok(())
 }
+
+#[pyclass(name = "Ballista", module = "ballista", subclass)]
+pub struct PyBallista(pub Option<HashMap<String, String>>);
+
+#[pymethods]
+impl PyBallista {
+    #[staticmethod]
+    #[pyo3(signature = (config=None))]
+    pub fn config(config: Option<HashMap<String, String>>) -> Self {
```

Comment:
it would be great if this method is overloaded, so it can accept single conf key and value

Reply:
Working on this, the problem is that `pyclass` doesn't accept generic type parameters, which makes it tricky because I can't just feed in `pub Pyballista<T>(pub Option<T>)` then write a `From` impl to convert a (&str, &str) into a HashMap. I guess I could just do a Vec<(&str, &str)>.

Reply:
I will actually just do that and see how you all like it instead of involving HashMap

Reply:
Actually can't use lifetimes in pyclass so using `String`s instead